From 59dde69334b351022cb60939c7de37145c4c4c54 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Wed, 31 Mar 2021 13:50:28 +0530
Subject: [PATCH 1387/1921] octeontx2-pf: Fix bandwidth profile
 NIX_AQ_INSTOP_WRITE

NIX admin quueue operation NIX_AQ_INSTOP_WRITE
performs a masked write to a NIX-managed hardware
structure. Software provides the data to be written
to the structure. Software must also provide a write
mask of the structure, where each bit to be updated
is a 1, and each bit not to be modified is a 0.
For configuring NIX ingress bandwidth profile context
the masks are missed. This patch fixes that.

Fixes: bd825b111cf3 ("octeontx2-pf: TC_MATCHALL ingress")
Change-Id: I640660e48b98902fe96101f852fde0a4402159fa
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48932
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/nic/cn10k.c  | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/cn10k.c b/drivers/net/ethernet/marvell/octeontx2/nic/cn10k.c
index 9be30a39d160..c40ea0386e3d 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/cn10k.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/cn10k.c
@@ -413,21 +413,38 @@ int cn10k_set_ipolicer_rate(struct otx2_nic *pfvf, u16 profile,
 
 	/* Set initial color mode to blind */
 	aq->prof.icolor = 0x03;
+	aq->prof_mask.icolor = 0x03;
 
 	/* Set rate and burst values */
 	aq->prof.cir_exponent = rate_exp;
+	aq->prof_mask.cir_exponent = 0x1F;
+
 	aq->prof.cir_mantissa = rate_mantissa;
+	aq->prof_mask.cir_mantissa = 0xFF;
+
 	aq->prof.cbs_exponent = burst_exp;
+	aq->prof_mask.cbs_exponent = 0x1F;
+
 	aq->prof.cbs_mantissa = burst_mantissa;
+	aq->prof_mask.cbs_mantissa = 0xFF;
+
 	aq->prof.rdiv = rdiv;
+	aq->prof_mask.rdiv = 0xF;
 
 	/* Two rate three color marker
 	 * With PEIR/EIR set to zero, color will be either green or red
 	 */
 	aq->prof.meter_algo = 2;
+	aq->prof_mask.meter_algo = 0x3;
+
 	aq->prof.rc_action = NIX_RX_BAND_PROF_ACTIONRESULT_DROP;
+	aq->prof_mask.rc_action = 0x3;
+
 	aq->prof.yc_action = NIX_RX_BAND_PROF_ACTIONRESULT_PASS;
+	aq->prof_mask.yc_action = 0x3;
+
 	aq->prof.gc_action = NIX_RX_BAND_PROF_ACTIONRESULT_PASS;
+	aq->prof_mask.gc_action = 0x3;
 
 	/* Fill AQ info */
 	aq->qidx = profile;
-- 
2.31.1

